lib/cocoapods/generator/xcconfig/pod_xcconfig.rb

```diff
@@ -44,14 +44,10 @@ def save_as(path)
         # @return [Xcodeproj::Config]
         #
         def generate
-          target_search_paths = target.build_headers.search_paths(target.platform)
```

Comment:
we no longer just add every single target header search paths. We instead figure out the search paths from dependent targets.